subsys/net/l2/ieee802154/ieee802154_mgmt.c

```diff
@@ -150,28 +157,30 @@ static int ieee802154_scan(uint32_t mgmt_request, struct net_if *iface,
 			if (ret) {
 				NET_DBG("Could not send Beacon Request (%d)",
 					ret);
-				net_pkt_unref(pkt);
```

Comment:
No, the packet got referenced above: line 153.

Reply:
This code block now calls the "out" label where the packet is being unrefed. Therefore this should be ok.

Reply:
No, it got referenced once more just 7 lines before, we need to get rid of this reference.
When the pkt gets created, its reference count has been initialized to 1. Here in this code block we put it to 2. 
If we specifically do not get back to a ref count of 1, at the out label it will get back to 1 and... the pkt will not be freed. (it needs to get to 0 for this to happen). So this pkt will be accounted as memory loss.

Reply:
As the comment was hidden I overlooked your response. Will fix this now.